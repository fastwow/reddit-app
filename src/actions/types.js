export const FETCH_POSTS_STARTED = 'fetch_posts_started';
export const FETCH_POSTS_SUCCESS = 'fetch_posts_success';
export const FETCH_MORE_POSTS_STARTED = 'fetch_more_posts_started';
export const FETCH_MORE_POSTS_SUCCESS = 'fetch_more_posts_success';
export const FETCH_REFRESHED_POSTS_STARTED = 'fetch_refreshed_posts_started';
export const FETCH_REFRESHED_POSTS_SUCCESS = 'fetch_refreshed_posts_success';
export const FETCH_FAVORITES_POSTS_STARTED = 'fetch_favorites_posts_started';
export const FETCH_FAVORITES_POSTS_SUCCESS = 'fetch_favorites_posts_success';
export const ADD_TO_FAVORITE_SUCCESS = 'add_to_favorite_success';
export const REMOVE_FROM_FAVORITE_SUCCESS = 'remove_from_favorite_success';
export const FILTER_POSTS_SUCCESS = 'search_post_success';
