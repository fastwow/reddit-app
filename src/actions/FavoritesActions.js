import {
  ADD_TO_FAVORITE_SUCCESS,
  REMOVE_FROM_FAVORITE_SUCCESS,
  FETCH_FAVORITES_POSTS_STARTED,
  FETCH_FAVORITES_POSTS_SUCCESS,
} from './types';
import {fetchAll, add, remove} from '../storage/favoritesStorage';

export const fetchFavorites = () => {
  return function (dispatch) {
    dispatch({type: FETCH_FAVORITES_POSTS_STARTED});
    return fetchAll()
      .then(res => {
        dispatch({
          type: FETCH_FAVORITES_POSTS_SUCCESS,
          data: res.data,
        });
      })
      .catch(error => {
        console.log(error);
      });
  };
};

export const addToFavorite = item => {
  return function (dispatch) {
    return add(item)
      .then(() => {
        dispatch({
          type: ADD_TO_FAVORITE_SUCCESS,
        });
      })
      .catch(error => {
        console.log(error);
      });
  };
};

export const removeFromFavorite = item => {
  return function (dispatch) {
    return remove(item)
      .then(() => {
        dispatch({
          type: REMOVE_FROM_FAVORITE_SUCCESS,
        });
      })
      .catch(error => {
        console.log(error);
      });
  };
};
