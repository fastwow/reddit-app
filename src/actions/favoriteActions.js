import {ADD_TO_FAVORITE_SUCCESS, REMOVE_FROM_FAVORITE_SUCCESS} from './types';

export function addToFavorite(item) {
  return {
    type: ADD_TO_FAVORITE_SUCCESS,
    item,
  };
}

export function removeFromFavorite(item) {
  return {
    type: REMOVE_FROM_FAVORITE_SUCCESS,
    item,
  };
}
