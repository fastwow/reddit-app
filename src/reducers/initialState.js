export default {
  posts: {isLoading: false, isRefreshing: false, errorMessage: '', posts: [], after: ''},
  favorites: {posts: []},
  filter: {searchTerm: ''},
};
