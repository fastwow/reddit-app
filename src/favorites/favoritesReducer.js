import initialState from './favoritesActionsInitialState';
import {ADD_TO_FAVORITE, REMOVE_FROM_FAVORITE} from './favoritesActionTypes';

export default function (state = initialState, action) {
  switch (action.type) {
    case ADD_TO_FAVORITE:
      return {
        ...state,
        posts: state.posts.concat(action.data),
      };
    case REMOVE_FROM_FAVORITE:
      return {
        ...state,
        posts: state.posts.filter(item => action.data.id !== item.id),
      };
    default:
      return state;
  }
}
