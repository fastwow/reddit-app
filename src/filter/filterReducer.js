import {FILTER_POSTS} from './filterActionTypes';
import initialState from './filterActionsInitialState';

export default function (state = initialState, action) {
  switch (action.type) {
    case FILTER_POSTS:
      return {
        ...state,
        searchTerm: action.searchTerm,
      };
    default:
      return state;
  }
}
