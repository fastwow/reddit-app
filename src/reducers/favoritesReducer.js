import initialState from './initialState';
import {ADD_TO_FAVORITE, REMOVE_FROM_FAVORITE} from '../actions/types';

export default function (state = initialState.favorites, action) {
  switch (action.type) {
    case ADD_TO_FAVORITE:
      return {
        ...state,
        posts: state.posts.concat(action.data),
      };
    case REMOVE_FROM_FAVORITE:
      return {
        ...state,
        posts: state.posts.filter(item => action.data.id !== item.id),
      };
    default:
      return state;
  }
}
