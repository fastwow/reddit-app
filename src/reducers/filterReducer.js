import {
  FILTER_POSTS,
} from '../actions/types';
import initialState from './initialState';

export default function (state = initialState.filter, action) {
  switch (action.type) {
    case FILTER_POSTS:
      return {
        ...state,
        searchTerm: action.searchTerm,
      };
    default:
      return state;
  }
}
