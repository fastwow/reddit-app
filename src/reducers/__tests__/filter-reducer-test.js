import initialState from '../initialState';
import filterReducer from '../filterReducer';
import {FILTER_POSTS} from '../../actions/types';
import favoritesReducer from '../favoritesReducer';

it('returns the same state for filter reducer on an unhandled action', () => {
  expect(filterReducer(initialState, {type: '_NULL'})).toMatchSnapshot();
});

it(`handles ${FILTER_POSTS} action`, () => {
  expect(favoritesReducer(initialState, {searchTerm: 'cat', type: FILTER_POSTS})).toMatchSnapshot();
});
