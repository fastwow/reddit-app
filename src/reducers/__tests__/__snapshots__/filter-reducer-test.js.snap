// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles filter_posts action 1`] = `
Object {
  "favorites": Object {
    "posts": Array [],
  },
  "filter": Object {
    "searchTerm": "",
  },
  "posts": Object {
    "after": "",
    "error": "",
    "isLoading": false,
    "isRefreshing": false,
    "posts": Array [],
  },
}
`;

exports[`returns the same state for filter reducer on an unhandled action 1`] = `
Object {
  "favorites": Object {
    "posts": Array [],
  },
  "filter": Object {
    "searchTerm": "",
  },
  "posts": Object {
    "after": "",
    "error": "",
    "isLoading": false,
    "isRefreshing": false,
    "posts": Array [],
  },
}
`;
